radixdlt/babylon-wallet-android
# [ABW-3763] - Account lockers

[Jira Ticket 1](https://radixdlt.atlassian.net/browse/ABW-3763)
[Jira Ticket 2](https://radixdlt.atlassian.net/browse/ABW-3757)
[Jira Ticket 3](https://radixdlt.atlassian.net/browse/ABW-3760)

## Description
This PR implements the account locker claim status check once in 5 minutes or whenever the user connects to a new dApp and it has a locker associated.


## How to test

In [this](https://rdxworks.slack.com/archives/C031A0V1A1W/p1724936298784399) thread you can find how to set up the environment to test the PR (I can run the airdrop manifest for you).

Claim status:
1. Run the airdrop manifest instructions using one of the test dApp accounts
2. Check that the locker deposit prompt is shown for the account in the wallet and the account details screens.
3. Check that the locker deposit label is shown for the dApp in the Authorized dApps screen

Show deposits switch:
1. Uncheck the deposits switch in dApp details screen while having some claimable resources from this dApp
2. Check that the deposit prompts are no longer displayed for the accounts linked to the dApp at point 1

Claim:
1. Run the airdrop manifest for fungibles
2. Tap on the deposit prompt
3. Check that the transaction review screen is shown with no errors
4. Sign the transaction
5. Wait for the transaction result and check if the deposit prompt was hidden
